jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -522,8 +524,9 @@ private void checkClassAccess(Class<?> targetClass) throws IllegalAccessExceptio
 				if (Modifier.isProtected(modifiers)) {
 					/* Interfaces are not classes but types, and should therefore not have access to 
 					 * protected methods in java.lang.Object as subclasses.
+					 * They should have access to members of the same package.
 					 */
-					if (!accessClass.isInterface()) {
+					if (!accessClass.isInterface() || isSamePackage(accessClass, targetClass)) {
```

Comment:
This needs more thought

Reply:
We may be excluding protected members from superinterfaces, right?

Reply:
> protected member interface foo.KlassA.InterfaceB.

Can you provide a test case?  I'm not 100% sure I know what a protected member interface is.

Reply:
```
package toe.foo;
public class Klass {
	protected interface PI {
		public default int getInt() {
		}
	}
}
```
`PI` is visible only to classes in package `toe.foo` or subclasses of `Klass`.

Reply:
I think the `protected` in this case is a Java Language Spec (JLS) statement.  The VM view of the world doesn't include protected classes - classes are either public or package in the VM.

If we remove this entire check for `protected` and instead treat it as `public` are there any issues with the existing tests?

Reply:
I am testing now, though this is a caution:
`Interfaces are not classes but types, and should therefore not have access to protected methods in java.lang.Object as subclasses.`

Reply:
@pdbain-ibm This is in `checkClassAccess()`.  There are other `checkAccess` methods for checking access to members.  Which leaves this check as a grey area that needs further investigation.

Thanks for running the extra tests.